src/Gateway.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace DigiTickets\OmnipayPay360HostedCashier;
+
+use DigiTickets\OmnipayPay360HostedCashier\Message\CompleteRedirectPurchaseRequest;
+use DigiTickets\OmnipayPay360HostedCashier\Message\RedirectPurchaseRequest;
+use DigiTickets\OmnipayPay360HostedCashier\Message\RefundRequest;
+use DigiTickets\OmnipayPay360HostedCashier\Traits\GatewayParamsTrait;
+use Omnipay\Common\AbstractGateway;
+use Omnipay\Common\Message\AbstractRequest;
+use Omnipay\Common\Message\RequestInterface;
+
+/**
+ * @method \Omnipay\Common\Message\RequestInterface authorize(array $options =[])
+ * @method \Omnipay\Common\Message\RequestInterface completeAuthorize(array$options = [])
+ * @method \Omnipay\Common\Message\RequestInterface capture(array $options =[])
+ * @method \Omnipay\Common\Message\RequestInterface void(array $options = [])
+ * @method \Omnipay\Common\Message\RequestInterface createCard(array $options =[])
+ * @method \Omnipay\Common\Message\RequestInterface updateCard(array $options =[])
+ * @method \Omnipay\Common\Message\RequestInterface deleteCard(array $options =[])
+ */
+class Gateway extends AbstractGateway
+{
+    use GatewayParamsTrait;
+
+    public function getName(): string
+    {
+        return 'Pay360HostedCashier';
+    }
+
+    public function getDefaultParameters(): array
+    {
+        return [
+            'installationId' => '',
+            'apiUsername' => '',
+            'apiPassword' => '',
+            'testMode' => false,
+        ];
+    }
+
+    /**
+     * Call when Pay360 sends a callback request to the merchant system (from notifyUrl param),
+     * and/or when Pay360 redirects back to the merchant site (from redirectUrl param).
+     * This validates that the payment was successful or was denied.
+     * @return AbstractRequest|CompleteRedirectPurchaseRequest
+     */
+    public function completePurchase(array $options = []): RequestInterface
+    {
+        $options = array_merge($this->getParameters(), $options);
+
+        return $this->createRequest(
+            CompleteRedirectPurchaseRequest::class,
+            $options
+        );
+    }
+
+    /**
+     * @return AbstractRequest|RedirectPurchaseRequest - provides a url to redirect to
```

Comment:
No, no! This is still not correct! This method returns a _request_ object. That object does not provide the redirect URL.

It is used when the merchant system makes the initial request to the gateway, and that request returns a _response_ object (specifically `RedirectPurchaseRequest`) and _that_ object has the redirect URL.

Sorry, it needs to be changed or it'll be confusing :-)

Reply:
The request does provide the ability to get URL :P Just trying to work out how to say to say that this purchase thing will eventually get you a URL to redirect to.
Oh well no comment is better than a bad comment!

Reply:
Well, yes, it kind of provides the _ability_ to get the URL, but it doesn't provide the URL directly.

I know why you're trying to say it, but I don't think you need to say anything here. The place to say that this is a redirect gateway is in the `readme` file. Anyone familiar with Omnipay will know that the Gateway class provides a purchase request, which is used to send the initial request, which returns a purchase response which says it's a redirect and provides the URL.